test/framework/common.go

```diff
@@ -17,30 +17,37 @@ import (
 
 // WildFlyBasicTest runs basic operator tests
 func WildFlyBasicTest(t *testing.T, applicationTag string) {
-	ctx := framework.NewTestCtx(t)
+	ctx, f := wildflyTestSetup(t)
 	defer ctx.Cleanup()
+
+	if err := wildflyBasicServerScaleTest(t, f, ctx, applicationTag); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func wildflyTestSetup(t *testing.T) (*framework.TestCtx, *framework.Framework) {
+	ctx := framework.NewTestCtx(t)
 	err := ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
 	if err != nil {
-		t.Fatalf("failed to initialize cluster resources: %v", err)
+		defer ctx.Cleanup()
```

Comment:
that does not look correct. If you are deferring the cleanup, it will be cleaned up when the method returns.

Reply:
This should be correct as the `defer.Cleanup()` is called just before the `t.Fatalf` is invoked. Saying differently it will be called when the method returns but the method returns just at the next call with error for whole test.